addons/medical_vitals/functions/fnc_updateOxygen.sqf

```diff
@@ -0,0 +1,81 @@
+#include "script_component.hpp"
+/*
+ * Author: Brett Mayson
+ * Update the oxygen levels
+ *
+ * Arguments:
+ * 0: The Unit <OBJECT>
+ * 1: Time since last update <NUMBER>
+ * 2: Sync value? <BOOL>
+ *
+ * ReturnValue:
+ * Current SPO2 <NUMBER>
+ *
+ * Example:
+ * [player, 1, false] call ace_medical_vitals_fnc_updateOxygen
+ *
+ * Public: No
+ */
+
+params ["_unit", "_deltaT", "_syncValue"];
+
+#define IDEAL_PPO2 0.255
+
+private _current = GET_SPO2(_unit);
+private _heartRate = GET_HEART_RATE(_unit);
+
+private _altitude = EGVAR(common,mapAltitude) + ((getPosASL _unit) select 2);
+private _po2 = if (missionNamespace getVariable ["ace_weather", false]) then {
+    private _temperature = _altitude call EFUNC(weather,calculateTemperatureAtHeight);
+    private _pressure = _altitude call EFUNC(weather,calculateBarometricPressure);
+    [_temperature, _pressure, EGVAR(weather,currentHumidity)] call EFUNC(weather,calculateOxygenDensity)
```

Comment:
Either just do `call EFUNC(weather,calculateAirDensity) * 0.21` here or change `calculateOxygenDensity` to take altitude as single param IMO. No sense in adding a function just to multiply by a constant if calling it needs this much setup.

Reply:
Not sre what you are suggesting about a single param

Reply:
Just pass `_altitude` and sweep the other calls under the rug.